### PR TITLE
Added parameter "apply_gradients" to disable parameter update step.

### PR DESCRIPTION
**Pull request by sfetzel**
If you want to use the preconditioning only and use another first order optimizer like Adam, you need to disable the parameter updates in the "step" function. I added the parameter "apply_gradients", which can be set to False to avoid the parameter updates.

The new training loop would look like this:
```python
kfac = torch_kfac.KFAC(model, learning_rate=1e-3, damping=1e-3,damping_adaptation_decay=0, momentum=0, apply_gradients=False)
optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=0.005)
...
model.zero_grad()
with kfac.track_forward():
  # forward pass
  loss = ...
with kfac.track_backward():
  # backward pass
  loss.backward()
kfac.step()
optimizer.step()
```
I don't know if this is actually necessary, because I could also set the learning rate to 0. What do you think?
Thanks a lot for the work on this library.

### CODE REVIEW

**Comment by n-gao** [comment]
Thanks a lot for the PR! :) As you correctly point out, setting the lr to 0 has an identical effect. Is there a particular reason we should add this as an extra flag instead of adding a comment to the learning rate?

**Comment by sfetzel** [comment]
You're right. I removed the new parameter again (see my last commit) and just added a check if the learning rate is 0. The reason for this check is that a multiplication with 0 costs the same amount of time as a multiplication with any other number, see this benchmark:

```python
import torch
import time

def measure_addition_time(vector_a, vector_b, warmup_runs=20, measured_runs=50):
    # Warm-up runs
    for _ in range(warmup_runs):
        _ = vector_a + vector_b
    
    # Measured runs
    start_time = time.time()
    for _ in range(measured_runs):
        _ = vector_a + vector_b
    end_time = time.time()
    
    # Calculate average time per addition
    average_time = (end_time - start_time) / measured_runs
    return average_time

# Define the size of the vectors
vector_size = 100000000

# Create two random vectors for the first scenario
vector1 = torch.randn(vector_size)
vector2 = torch.randn(vector_size)

# Create a zero vector for the second scenario
zero_vector = torch.zeros(vector_size)

# Measure time for adding two non-zero vectors with warm-up
time_non_zero_addition = measure_addition_time(vector1, vector2)
print(f"Average time for adding two non-zero vectors (with warm-up): {time_non_zero_addition} seconds")

# Measure time for adding a non-zero vector and a zero vector with warm-up
time_zero_addition = measure_addition_time(vector1, zero_vector)
print(f"Average time for adding a non-zero vector and a zero vector (with warm-up): {time_zero_addition} seconds")
```

**Comment by n-gao** [comment]
Sure that makes sense, even though parameter updates can usually be neglected as computational cost.
